### PR TITLE
CAMEL-19526: replaced the Thread.sleep with the best option or removed it.

### PR DESCRIPTION
**Pull request by rafamqrs**
# Description

Replaced Thread.sleep() in camel-jms.

# Target

- [X] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [X] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [X] I checked that each commit in the pull request has a meaningful subject line and body.

- [X] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

**Review by oscerd — APPROVED** [review]
cc @orpiske

**Comment by orpiske** [comment]
Thanks for the contribution! I'll be looking closely at this one later today ... currently busy with a CI issue.

**Review by orpiske — CHANGES_REQUESTED** [review]
Just marking like this 'cause it's pending further review.

File: components/camel-jms/src/test/java/org/apache/camel/component/jms/async/MyAsyncProducer.java [hunk 1]
```diff
@@ -51,7 +53,10 @@ public boolean process(final Exchange exchange, final AsyncCallback callback) {
         executor.submit(() -> {
 
             LOG.info("Simulating a task which takes {} millis to reply", getEndpoint().getDelay());
-            Thread.sleep(getEndpoint().getDelay());
+
+            Awaitility.await()
+                    .pollDelay(Duration.ofMillis(getEndpoint().getDelay()))
+                    .until(() -> true);
```
Comment (orpiske) [inline components/camel-jms/src/test/java/org/apache/camel/component/jms/async/MyAsyncProducer.java:59]: I think Thread.sleep is fine here.

File: components/camel-jms/src/test/java/org/apache/camel/component/jms/issues/TempReplyToIssueTest.java [hunk 2]
```diff
@@ -75,7 +78,9 @@ public String handleMessage(
         producer.stop();
 
         // sleep a bit so Camel will send the reply a bit later
-        Thread.sleep(1000);
+        Awaitility.await()
+                .pollDelay(Duration.ofMillis(1000))
+                .until(() -> true);
```
Comment (orpiske) [inline components/camel-jms/src/test/java/org/apache/camel/component/jms/issues/TempReplyToIssueTest.java:83]: I think `Thread.sleep` is fine here.

File: components/camel-jms/src/test/java/org/apache/camel/component/jms/temp/JmsReconnectManualTest.java [hunk 3]
```diff
@@ -102,7 +105,9 @@ public void configure() {
          * Wait long enough for the jms client to do a full reconnect. In the Tibco EMS case this means that a Temporary
          * Destination created before is invalid now
          */
-        Thread.sleep(5000);
+        Awaitility.await()
+                .pollDelay(Duration.ofMillis(5000))
+                .until(() -> true);
```
Comment (orpiske) [inline components/camel-jms/src/test/java/org/apache/camel/component/jms/temp/JmsReconnectManualTest.java:110]: I think `Thread.sleep` is fine here.

**Review by orpiske — CHANGES_REQUESTED** [review]
Thanks for your contribution!

There are few places where I think it would be better to retain the `Thread.sleep` because the proposed solution with Awaitility makes the code harder to read (and, also, the adequate solution would be to reword the tests so that we can use adequate synchronization mechanisms - but that's beyond the point of this change).

Also, one more note about the commit message. Please make sure to write the commit message according to our standards. Trying doing something like this: 

```
CAMEL-19526: replaced the thread.sleep with the best option or removed
```

File: components/camel-jms/src/test/java/org/apache/camel/component/jms/temp/JmsReconnectManualTest.java (hunk 3 above)
Comment (rafamqrs) [reply components/camel-jms/src/test/java/org/apache/camel/component/jms/temp/JmsReconnectManualTest.java]: Makes Sense, thank you!

**Comment by rafamqrs** [comment]
> Thanks for your contribution!
> 
> There are few places where I think it would be better to retain the `Thread.sleep` because the proposed solution with Awaitility makes the code harder to read (and, also, the adequate solution would be to refactor the tests so that we can use adequate synchronization mechanisms - but that's beyond the point of this change).
> 
> Also, one more note about the commit message. Please make sure to write the commit message according to our standards. Trying doing something like this:
> 
> ```
> CAMEL-19526: replaced the thread.sleep with the best option or removed
> ```

My bad, I ended up forgetting to write the Jira task on the commit message and keep the commit message according to the standards.

**Review by orpiske — APPROVED** [review]
Thanks for the contribution! LGTM

**Pull request merged**